jetty-servlet-api/pom.xml

```diff
@@ -175,97 +192,110 @@
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <version>3.0.1</version>
+        <executions>
+          <execution>
+            <id>attach-javadocs</id>
+            <goals>
+              <goal>jar</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <tags>
+            <tag>
+              <name>apiNote</name>
+              <placement>a</placement>
+              <head>API Note:</head>
+            </tag>
+            <tag>
+              <name>implSpec</name>
+              <placement>a</placement>
+              <head>Implementation Requirements:</head>
+            </tag>
+            <tag>
+              <name>implNote</name>
+              <placement>a</placement>
+              <head>Implementation Note:</head>
+            </tag>
+          </tags>
+        </configuration>
+      </plugin>
 
       <!-- generate the manifest for the classes jar -->
       <plugin>
-          <groupId>org.apache.felix</groupId>
-          <artifactId>maven-bundle-plugin</artifactId>
-          <version>3.5.1</version>
-          <extensions>true</extensions>
-          <executions>
-              <execution>
-                  <id>bundle-manifest</id>
-                  <goals>
-                      <goal>manifest</goal>
-                  </goals>
-              </execution>
-          </executions>
-          <configuration>
-              <instructions>
-                  <Bundle-SymbolicName>org.eclipse.jetty.osgi-servlet-api</Bundle-SymbolicName>
-                  <Bundle-Name>Jetty Servlet API and Schemas for OSGi</Bundle-Name>
-                  <Bundle-Description>Combined servlet api and schemas for use in osgi environments</Bundle-Description>
-                  <Bundle-RequiredExecutionEnvironment>JavaSE-1.7</Bundle-RequiredExecutionEnvironment>
-                  <_nouses>true</_nouses>
-                  <Export-Package>!javax.servlet.jsp.*,javax.servlet.*;version="4.0",*</Export-Package>
-              </instructions>
-          </configuration>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <version>3.5.1</version>
+        <extensions>true</extensions>
+        <executions>
+          <execution>
+            <id>bundle-manifest</id>
+            <goals>
+              <goal>manifest</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <instructions>
+            <Automatic-Module-Name>jetty-servlet-api</Automatic-Module-Name>
+            <Bundle-SymbolicName>org.eclipse.jetty.osgi-servlet-api</Bundle-SymbolicName>
+            <Bundle-Name>Jetty Servlet API and Schemas for JPMS and OSGi</Bundle-Name>
+            <Bundle-Description>Combined servlet api and schemas for use in JPMS and OSGi environments
+            </Bundle-Description>
+            <Bundle-RequiredExecutionEnvironment>JavaSE-1.7</Bundle-RequiredExecutionEnvironment>
```

Comment:
As I said in #20, Servlet 4.0 requires Java 8.

Reply:
@janbartel ping, are you ok with upgrading this to `JavaSE-1.8`? (this will impact OSGi users of Jetty 10.x)

Reply:
This would be a repackaging, not a recompile.

Checking the bytecode ...

```
$ javap -classpath ~/.m2/repository/javax/servlet/javax.servlet-api/4.0.0/javax.servlet-api-4.0.0.jar \
   -verbose javax.servlet.http.HttpServletRequest | head -10
Classfile jar:file:///home/joakim/.m2/repository/javax/servlet/javax.servlet-api/4.0.0/javax.servlet-api-4.0.0.jar!/javax/servlet/http/HttpServletRequest.class
  Last modified Aug 15, 2017; size 3054 bytes
  MD5 checksum 2c69ad40199e2a175050c78bdbf5586b
  Compiled from "HttpServletRequest.java"
public interface javax.servlet.http.HttpServletRequest extends javax.servlet.ServletRequest
  minor version: 0
  major version: 52
  flags: (0x0601) ACC_PUBLIC, ACC_INTERFACE, ACC_ABSTRACT
  this_class: #4                          // javax/servlet/http/HttpServletRequest
  super_class: #5                         // java/lang/Object
$
```

Yup, `major version: 52` would be bytecode for Java 8.

Reply:
@joakime  I guess its ok to upgrade to 1.8 for Bundle-RequiredExecutionEnvironment, as that is the default already for jetty-9.4.x, and it only signifies the minimum version.

BTW, the pom.xml for jetty-10.0.x says JavaSE-1.8 - I assume that should be set to something much higher?